xds/internal/xdsclient/authority.go

```diff
@@ -633,14 +633,20 @@ func (a *authority) watchResource(rType xdsresource.Type, resourceName string, w
 		// Always add the new watcher to the set of watchers.
 		state.watchers[watcher] = true
 
-		// If we have a cached copy of the resource, notify the new watcher.
+		// If we have a cached copy of the resource, notify the new watcher
+		// immediately.
 		if state.cache != nil {
 			if a.logger.V(2) {
 				a.logger.Infof("Resource type %q with resource name %q found in cache: %s", rType.TypeName(), resourceName, state.cache.ToJSON())
 			}
 			resource := state.cache
 			a.watcherCallbackSerializer.TrySchedule(func(context.Context) { watcher.OnUpdate(resource, func() {}) })
 		}
+		// If last update was NACK'd, notify the new watcher of error
+		// immediately as well.
+		if state.md.Status == xdsresource.ServiceStatusNACKed && state.md.ErrState != nil {
```

Comment:
Is it not enough if we check for `state.md.Status == xdsresource.ServiceStatusNACKed` alone? There is a single place where the status is set to `xdsresource.ServiceStatusNACKed`, and in there, we always set the error field as well.

Reply:
yeah i checked that and its only at one place where we do populate the error